scripts/india_wris/base.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pandas as pd
+import json
+import re
+from india.geo.districts import IndiaDistrictsMapper
+
+
+class WaterQualityBase():
+
+    def __init__(self, dataset_name, util_names, template_strings):
+        self.dataset_name = dataset_name
+        self.util_names = util_names
+        self.module_dir = os.path.dirname(__file__)
+
+        self.solute_mcf = template_strings['solute_mcf']
+        self.solute_tmcf = template_strings['solute_tmcf']
+        self.chemprop_mcf = template_strings['chemprop_mcf']
+        self.chemprop_tmcf = template_strings['chemprop_tmcf']
+        self.site_dcid = template_strings['site_dcid']
+        self.unit_node = template_strings['unit_node']
+
+    def create_dcids_in_csv(self):
+        """
+        Method to map the district names to LGD District Codes
+        Mapped codes are used to create dcids for water quality stations
+        
+        Format of dcid for a station:
+            'india_wris/<lgd_code_of_district>_<name_of_station>'
+        Example: 'india_wris/579_Velanganni' for Velanganni station in
+                    Nagercoil, Tamil Nadu, India
+        """
+
+        self.df = pd.read_csv(
+            os.path.join(self.module_dir,
+                         'data/{}.csv'.format(self.util_names)))
+        # Dropping rows with all data missing
+        self.df.dropna(thresh=11, inplace=True)
+
+        # Mapping district names to LGD Codes
+        mapper = IndiaDistrictsMapper()
+        df_map = self.df[['StateName',
+                          'DistrictName']].drop_duplicates().dropna()
+        df_map['DistrictCode'] = df_map.apply(
+            lambda x: mapper.get_district_name_to_lgd_code_mapping(
+                x['StateName'], x['DistrictName']),
+            axis=1)
+
+        # Merging LGD codes with original df and creating dcids
+        self.df = self.df.merge(df_map.drop('StateName', axis=1),
+                                on='DistrictName',
+                                how='left')
+        self.df['dcid'] = self.df.apply(lambda x: ''.join([
+            'india_wris/',
+            str(x['DistrictCode']), '_', ''.join(
```

Comment:
Are Station Names guaranteed to be unique within districts, or is there another identifier that can be added to guarantee uniqueness in these generated ID's.

Reply:
Yes, the station names are unique. I confirmed this with the lat long. All stations had unique lat long values.

Reply:
sounds good.